managing-os/windows/windows-vm-test.md

```diff
@@ -0,0 +1,306 @@
+---
+layout: doc
+title: Installing a Windows VM
+permalink: /doc/windows-vm/
+---
+
+
+Installing a Windows VM
+=======================
+
+Qubes 4.0 - importing a Windows VM from R3.2
+-------------------------------------------
+
+Importing should work, simply make sure that you are not using Xen's newer linux stubdomain and that the VM is in HVM mode (these steps should be done automatically when importing the VM):
+
+~~~
+qvm-features VMNAME linux-stubdom ''
+qvm-prefs VMNAME virt_mode hvm
+~~~
+
+Note however that you are better off creating a new Windows VM to benefit from the more recent emulated hardware: R3.2 uses a MiniOS based stubdomain with an old and mostly unmaintained 'qemu-traditional' while R4.0 uses a Linux based stubdomain with a recent version of upstream qemu (see [this post](https://groups.google.com/d/msg/qubes-devel/tBqwJmOAJ94/xmFCGJnuAwAJ)).
+
+
+Qubes 3.2 - Windows VM installation
+-----------------------------------
+
+### Summary ###
+
+~~~
+qvm-create win7new --hvm --label red
+qvm-prefs -s win7new memory 4096
+qvm-prefs -s win7new maxmem 4096
+qvm-grow-root win7new 25g 
+qvm-prefs -s win7new debug true
+cp /var/lib/qubes/appvms/win7new/win7new.conf /tmp
+sed -i "s/<model \+type='xen' \+vram=/<model type='cirrus' vram=/" /var/lib/qubes/appvms/win7new/win7new.conf
+qvm-start --cdrom=untrusted:/home/user/windows_install.iso win7new
```

Comment:
Shouldn't this use copied config in /tmp (qvm-start --custom-config=...)? Also sed should operate on that config copy. The `/var/lib/qubes/appvms/win7new/win7new.conf` config is regenerated before each VM start.

Reply:
> Shouldn't this use copied config in /tmp (qvm-start --custom-config=...)?

No: the file copied to /tmp is a backup that we're supposed to restore later on with `cp /tmp/win7new.conf /var/...` ; but given your comment below I see that's not a good idea.

> The /var/lib/qubes/appvms/win7new/win7new.conf config is regenerated before each VM start.

I didn't know that. Out of curiosity, where does Qubes get the VM's "real" settings from then ? I don't have R3.2 anymore to test unfortunately....

Anyway, I've updated the doc to use `qvm-start --custom-config=` ; I initially wanted to avoid that because users might forget about it and only run `qvm-start` but I guess they'll have to stick to the doc :)

Hopefully I didn't forget anything (+ fixed the `tail -f` remark).

Reply:
> I didn't know that. Out of curiosity, where does Qubes get the VM's "real" settings from then ? 

From qubes.xml, basically what you see in qvm-prefs. In fact the .conf file in VM's directory isn't used at all, qvm-start write there the same thing that is sent to libvirt, just to ease modification. And unless you specify --custom-config, with a _different path_, actual config is built on each VM start.

Reply:
Thanks for the clarification.
The last version of the doc should be OK then (for 3.2, the instructions use `qvm-start --custom-config=/tmp/vmname.conf`, where the conf file in /tmp has been modified to use the cirrus video adapter).